crates/winmd/src/types/debug.rs

```diff
@@ -0,0 +1,60 @@
+use crate::types::RequiredInterface;
+use crate::types::TypeName;
+use proc_macro2::TokenStream;
+use quote::quote;
+
+pub fn debug_tokens(name: &TypeName, interfaces: &Vec<RequiredInterface>) -> TokenStream {
+    let constraints = name.constraints();
```

Comment:
Since the caller of debug_tokens already has the constraints, perhaps we can just pass it in to avoid recomputing.

Reply:
Definitely more tedius to do this, but if we think it will make a perf difference I can do it.

Reply:
Yes, since there's potentially tens of thousands of interfaces. Welcome suggestions for making this less tedious. We can also profile - not sure how expensive it is to compute the token stream.